tests/test_base.py

```diff
@@ -9,6 +9,14 @@
 SO_SUCCESS = 0
 
 
+class UnicodeErrorRaiserIO(BytesIO):
+    def readlines(self, *args, **kwargs):
+        for line in super().readlines(*args, **kwargs):
+            yield line.decode('utf-8')
+
+    __iter__ = readlines
+
```

Comment:
`super().__iter__()` (or variants thereof)  won't work, therefore we keep the `readlines` implementation.